tls/extensions/s2n_client_key_share.c

```diff
@@ -120,9 +123,17 @@ int s2n_extensions_client_key_share_recv(struct s2n_connection *conn, struct s2n
             /* Ignore curves with points we can't parse */
             conn->secure.client_ecc_evp_params[supported_curve_index].negotiated_curve = NULL;
             GUARD(s2n_ecc_evp_params_free(&conn->secure.client_ecc_evp_params[supported_curve_index]));
+        } else {
+            match = 1;
```

Comment:
Just a note; once my code change is made, this match flag is no longer needed since we would already be checking if we need to do a hrr after all the extensions have been parsed.